ghost/ghost/src/core/activitypub/http-signature.service.ts

```diff
@@ -0,0 +1,156 @@
+                        ...data
+                    };
+                } catch (err) {
+                    return data;
+                }
+            }, {});
+
+        if (
+            !signatureData.signature ||
+            !signatureData.headers ||
+            !signatureData.keyId ||
+            !signatureData.algorithm
+        ) {
+            throw new Error('Could not parse signature');
+        }
+
+        return {
+            keyId: new URL(signatureData.keyId),
+            headers: signatureData.headers.split(/\s/),
+            signature: Buffer.from(signatureData.signature, 'base64url'),
+            algorithm: signatureData.algorithm
+        };
+    }
+
+    private getHeader(
+        header: string,
+        headers: Record<string, string>,
+        requestMethod: string,
+        requestUrl: string
+    ) {
+        if (header === '(request-target)') {
+            return `${requestMethod.toLowerCase()} ${requestUrl}`;
+        }
+        if (!headers[header]) {
+            throw new Error(`Missing Header ${header}`);
+        }
+        return headers[header];
+    }
+
+    protected async getPublicKey(keyId: URL): Promise<crypto.KeyObject> {
+        try {
+            const keyRes = await fetch(keyId, {
+                headers: {
+                    Accept: 'application/ld+json'
+                }
+            });
+
+            // This whole thing is wrapped in try/catch so we can just cast as we want and not worry about errors
+            const json = (await keyRes.json()) as {publicKey: {publicKeyPem: string}};
+
+            console.log('KEY FOR FIXTURES');
+            console.log(json);
+            // ????
+            // const key = crypto.createPublicKey({key: json.publicKey?.publicKeyPem, type: 'pkcs1'});
+            const key = crypto.createPublicKey(json.publicKey.publicKeyPem);
+            return key;
+        } catch (err) {
+            throw new Error(`Could not find public key ${keyId.href}: ${err}`);
+        }
+    }
+
+    private validateDigest(signatureData: Signature, requestBody: Buffer, requestHeaders: Record<string, string>) {
+        const digest = crypto
+            .createHash(signatureData.algorithm)
+            .update(requestBody)
+            .digest('base64');
+
+        const remoteDigest = requestHeaders.digest?.split('SHA-256=')[1];
+
+        return digest === remoteDigest;
+    }
+
+    async validate(
+        requestMethod: string,
+        requestUrl: string,
+        requestHeaders: Record<string, string>,
+        requestBody: Buffer = Buffer.alloc(0, 0)
+    ) {
+        console.log('STUFF FOR FIXTURES');
+        console.log({
+            requestMethod,
+            requestUrl,
+            requestHeaders,
+            requestBody: requestBody.toString('base64')
+        });
+        const signatureData = this.parseSignatureHeader(
+            requestHeaders.signature
+        );
+
+        if (requestMethod === 'post') {
```

Comment:
This block is not being called by the tests because of a case mismatch "post" vs "POST"